### PR TITLE
Support Salesforce LWC HTML Shadow DOM web component recording

### PR DESCRIPTION
**Pull request by Vadman97**
## Summary

Brings in changes from 
https://github.com/highlight/rrweb/pull/118
to support building Salesforce LWC web components.

## How did you test this change?

https://app.highlight.io/59983/sessions/m115TBoiBhoRviembhTz5mQpF6ZR?page=1&query=

adding new [docs](https://highlight-landing-git-sup-15-investigate-s-f9489c-highlight-run.vercel.app/docs/getting-started/client-sdk/replay-configuration/salesforce-lwc) on LWC

## Are there any deployment considerations?

changeset

## Does this work require review from our design team?

no

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -76,9 +78,9 @@
 		"test:all": "yarn turbo run test --filter=!render --filter=!rrweb --filter=!@rrweb/types --filter=!rrweb-snapshot --filter=!rrweb-player --filter=!rrdom --filter=!rrdom-nodejs --filter=!nextjs",
 		"test:render": "yarn turbo run test --filter=render",
 		"sourcemaps:frontend": "yarn workspace @highlight-run/frontend sourcemaps",
-		"knip:frontend": "knip --workspace frontend"
+		"knip:frontend": "knip --workspace @highlight-run/frontend"
 	},
-	"devDependencies": {
+	"dependencies": {
```
Comment (ccschmitz) [inline package.json:83]: Just curious, what's the reason for this change?

**Review by ccschmitz — APPROVED** [review]
Code looks good to me + I ran the app locallly (using `dev:frontend`) and all looked good there in a quick click test too!

File: package.json (hunk 1 above)
Comment (Vadman97) [reply package.json:83]: was trying to figure out why `turbo` added as a top level `dev` dependency was not accessible to sub projects (ie. rrweb, having to add `turbo` to each subpackage). this didn't help tho so i'll undo it
